src/server/main_service.cc

```diff
@@ -1632,10 +1632,10 @@ void Service::EvalInternal(CmdArgList args, const EvalArgs& eval_args, Interpret
     });
 
     ++ServerState::tlocal()->stats.eval_shardlocal_coordination_cnt;
-    boost::intrusive_ptr<Transaction> stub_tx = new Transaction{tx};
+    boost::intrusive_ptr<Transaction> stub_tx = new Transaction{tx, *sid};
     cntx->transaction = stub_tx.get();
 
-    tx->PrepareMultiForScheduleSingleHop(*sid, 0, args);
+    tx->PrepareMultiForScheduleSingleHop(*sid, tx->GetDbIndex(), args);
```

Comment:
This really looked like a potential bug that nobody noticed 😜

Reply:
Nice catch :)

Please merge that, as that's the right thing to do, but generally we should not allow to SELECT other databases when in emulated cluster mode (we already [disallow](https://github.com/dragonflydb/dragonfly/blob/6a75c6ddc572cd75eacf0b365bee9e0f547da26a/src/server/generic_family.cc#L1158) that in real cluster)

Reply:
BTW of course this can be called without emulated cluster mode, so indeed good catch and please merge as I said!